### PR TITLE
map directory read error to empty directory

### LINKED ISSUE #10629: Error importing Lia
#### A non-readable but "executable" directory causes compilation failure when trying to import Lia.

OS: linux

```
$ mkdir a && chmod 0100 a && echo "Require Lia." > bug.v && PATH="$PATH":"`pwd`"/a coqc bug.v

File "./bug.v", line 1, characters 0-12:
Error: System error: "/tmp/a/.: Permission denied"
```

### PR DESCRIPTION
**Pull request by spanjel**
bug fix 

Fixes #10629 

(re-open of PR #10630)

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]
LGTM.

@vbgl you may consider this for backport, I didn't milestone 8.10 due to how late it is in the release process.

File: lib/system.ml [hunk 1]
```diff
@@ -53,8 +53,14 @@ module StrSet = Set.Make(StrMod)
 let dirmap = ref StrMap.empty
 
 let make_dir_table dir =
+  let entries =
+    try
+      Sys.readdir dir
+    with Sys_error _ ->
+      warn_cannot_open_dir dir;
+      Array.of_list [] in
```
Comment (vbgl) [inline lib/system.ml]: Why “Array.of_list []” rather than the empty array literal “[||]”?

**Comment by vbgl** [comment]
This should be fine for 8.10. Feel free to change the milestone when merging.

File: lib/system.ml (hunk 1 above)
Comment (ejgallego) [reply lib/system.ml]: Yup, good point, @spanjel , care to amend that part?

**Comment by spanjel** [comment]
my pidgin ocaml did not include that syntax :)

**Comment by ejgallego** [comment]
Looks good @spanjel thanks, I'll merge tomorrow.

In the future [or even now if you feel like it] I suggest you amend to the original commit, or squash the second one. IMHO no point in having the intermediate state in git history, and indeed given the size and long history of Coq we try to keep history a bit tidy.

**Comment by spanjel** [comment]
squashed in #10639

**Comment by Zimmi48** [comment]
Note that you can use `git rebase -i` to amend the history, then `git push` with the `--force` option to avoid having to recreate a branch every time you amend the history.

**Comment by ejgallego** [comment]
Indeed @spanjel  when modifying a PR we just use `--force-with-lease` so you can rewrite the PR without having to close it.

**Pull request closed without merging**